### PR TITLE
Remove deprecated modules: iris.unit, iris.proxy and iris.fileformats.pp_packing

### LINKED ISSUE #2642: Remove deprecated lib/iris/fileformats/pp_packing.py
The module `lib/iris/fileformats/pp_packing.py` was deprecated in Iris 1.10

Replaced by mo_pack.

https://github.com/SciTools/iris/blob/master/lib/iris/fileformats/pp_packing.py#L21

### LINKED ISSUE #2643: Remove deprecated lib/iris/proxy.py
The module `lib/iris/proxy.py` was deprecated in Iris 1.9

Replacement: use lazy imports

https://github.com/SciTools/iris/blob/master/lib/iris/proxy.py#L18

### LINKED ISSUE #2645: Remove deprecated lib/iris/unit.py
The module `lib/iris/unit.py` was deprecated in Iris 1.9.

Replacement: use cf_units

https://github.com/SciTools/iris/blob/master/lib/iris/unit.py#L18

### PR DESCRIPTION
**Pull request by lbdreyer**
Fixes #2642 
Fixes #2643 
Fixes #2645

### CODE REVIEW

**Comment by dkillick** [comment]
@lbdreyer I've modified the description for this PR - you'd linked #2642 twice and I think the second link was supposed to point to #2645.

File: setup.py [hunk 1]
```diff
@@ -210,18 +210,6 @@ def extract_version():
...
-            standard=False,
-            ext_modules=[
-                setuptools.Extension(
-                    libraries=['mo_unpack'],
-                    include_dirs=[np.get_include()]
-                )
-            ]
-        )
-    },
     cmdclass={'test': SetupTestRunner, 'build_py': BuildPyWithExtras,
```
Comment (corinnebosley) [inline setup.py]: @lbdreyer I take it this is only used for pp-packing?  Nothing else will be affected by the removal of this code section?

**Comment by corinnebosley** [comment]
Pending clarification from @lbdreyer, I am happy to merge this.  All changes look good and all tests pass.

**Comment by pelson** [comment]
Duped in #2363.

**Pull request merged**